app/systemd/entry.sh

```diff
@@ -85,7 +90,13 @@ function init_systemd()
 	EOF
 
  	sleep infinity &
-	exec env DBUS_SYSTEM_BUS_ADDRESS=unix:path=/run/dbus/system_bus_socket /sbin/init quiet systemd.show_status=0
+
+	if version_gt ${BALENA_HOST_OS_VERSION##* } 2.41.1; then
```

Comment:
One think I'm wondering, what happens when this parameter is not defined (e.g. if not running inside a balenaOS device started up by the container). What would be the better fallback at the moment?

Currently if `BALENA_HOST_OS_VERSION` is not defined, it will fall back to the legacy cgroup controller. Is that the right choice?

Would there be any other way to establish whether or not the an environment can run without the legacy thing? For example does the systemd version matter? Or the host systemd / container systemd interaction is what matter? 

Basically trying to establish a bit what would be a good default, and the narrowest circumstances when the legacy type *must* be used.

In general I think it's a good PR to have, just trying to figure out whether this is the best form we can make it at the moment?

Reply:
Agree, @ZubairLK is there something else we can check instead of this env var?